### ISSUE TITLE
Unable to create a new SPT with token and secret

### ISSUE THREAD

**Issue by dramich** [Azure/go-autorest#283]
Currently there is no way to create a new SPT if you have an existing token from a Web app and a secret. 

Use case: Storing the token and the required SPT config details (urls, app secret...) in a database (etcd) and be able to recover from a system crash without requiring a user to login again and proceed through the code auth flow with AzureAD.

**Comment by jhendrixMSFT** [comment]
Interesting idea.  Would adding the necessary marshaller/unmarshaller to the token type work for you?

**Comment by dramich** [comment]
I don't think that would get me back to a working `ServicePrincipalToken`. A setter on the SPT allowing setting of a secret or token would also work.

**Comment by jhendrixMSFT** [comment]
I whipped up a prototype for marshalling/unmarshalling a SPT and was able to convert a SPT to/from JSON before creating a redis cache.  Take a look at https://github.com/jhendrixMSFT/go-autorest/commit/de00837dae175d87f42c5f0ed0e3394970a5e233, still a little rough but you get the idea.

**Comment by dramich** [comment]
Ah, when you said token you were referring to the SPT. This looks like it would work great.
